src/app/components/authorized/subreddit/subreddit.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { SubredditService } from 'src/app/services/subreddit.service';
+import { Subreddit } from 'src/types/subreddits';
+
+@Component({
+    selector: 'app-subreddit',
+    templateUrl: './subreddit.component.html',
+    styleUrls: ['./subreddit.component.scss'],
+})
+export class SubredditComponent implements OnInit {
+    @Input() subredditName: string;
+
+    subreddit: Subreddit;
+
+    constructor(private _subredditService: SubredditService) {}
+
+    ngOnInit(): void {
+        // this.subreddit = this._subredditService.getSubreddit(this.subredditName);
```

Comment:
Commented code should not be part of code review as it will almost always not pass

Reply:
I deleted the content of subreddit and subreddits components since I haven't solved their correct functionality yet and it's not the goal of this ticket. I will fix it in the ticket where it's supposed to be.